packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1033,11 +1045,11 @@ class PopupMenuButton<T> extends StatefulWidget {
   /// and the button will behave like an [IconButton].
   final Widget? icon;
 
-  /// The offset applied to the Popup Menu Button.
+  /// The offset applied to the popup menu.
   ///
-  /// When not set, the Popup Menu Button will be positioned directly next to
+  /// When not set, the popup menu will be positioned directly over
   /// the button that was used to create it.
-  final Offset offset;
+  final Offset? offset;
```

Comment:
Instead of making this nullable, why not make it be relative to the position described by `position`?  Then, position can also be non-nullable, and if it defaults to `MenuPosition.over`, will not be a breaking change, since an offset of zero (the default), and the default position will be identical to the previous behavior.

Reply:
Thanks so much for the review. I did revert offset being nullable, however, I think the new position parameter should be nullable so that the user doesn't try to change offset to a random value when the position still has a fixed menu offset defined.

They can either just use the widget as it is, override position, or set an offset. Without conflicting fixed positions and a custom offset. If the default position says over but offset is Offset(-100,-100), it would be conflicting information.
Let me know if you think this makes sense.

Reply:
Sorry, I'm not sure I understand...

Why can't you add the offset to the default position? If the `position` is `over`, and the `offset` is zero, you have a location for the menu (the current default location before this PR).  If you then change the `offset`, it starts at that location, and modifies it by the `offset`.  If the position is `under`, and the `offset` is zero, then that also defines a location for the menu right below the button, and changing the `offset` will just modify the position relative to that point.

It doesn't sound conflicting to me, but maybe I've totally missed your point... Unless you just mean that "over" implies that it is visually on top of the button, but a large offset will mean that it is no longer visually over it? I think that's OK: they said they wanted it a long way away from the "over" location: it should be expected.

Reply:
Thanks so much for the explanation.

My idea was offset should override both positions enum but after reading our comment. I realized you wanted offset to apply on top of predefined offset and if the position is `under`, `offset` will just modify the position relative to that point.


I added conditions that handle this logic and expanded the test to cover offset changes.

Reply:
cc: @gspencergoog